src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -1268,6 +1525,213 @@ input ParkingFilter {
+    grouping off.
+    """
+    groupedOtherThanSameLegsMaxCostMultiplier: Float = 2.0
+}
+
+"""
+Plan location settings. Location must be set. Label is optional
+and used for naming the location.
+"""
+input PlanLabeledLocationInput {
+    """
+    A location that has to be used in an itinerary.
+    """
+    location: PlanLocationInput!
+
+    """
+    A label that can be attached to the location. This label is then returned with the location
+    in the itineraries.
+    """
+    label: String
+}
+
+"""
+Plan location. Either a coordinate or a stop location should be defined.
+"""
+input PlanLocationInput @oneOf {
+    """
+    Coordinate of the location. Note, either a coordinate or a stop location should be defined.
+    """
+    coordinate: PlanCoordinateInput
+
+    """
+    Stop, station, a group of stop places or multimodal stop place that should be used as
+    a location for the search. The trip doesn't have to use the given stop location for a
+    transit connection as it's possible to start walking to another stop from the given
+    location. If a station or a group of stop places is provided, a stop that makes the most
+    sense for the journey is picked as the location within the station or group of stop places.
+    """
+    stopLocation: PlanStopLocationInput
+}
+
+"""
+Wrapper type for different types of preferences related to plan query.
+"""
+input PlanPreferencesInput {
+    """
+    Street routing preferences used for ingress, egress and transfers. These do not directly affect
+    the transit legs but can change how preferable walking or cycling, for example, is compared to
+    transit.
+    """
+    street: PlanStreetPreferencesInput
+
+    """
+    Transit routing preferences used for transit legs.
+    """
+    transit: TransitPreferencesInput
+
+    """
+    Accessibility preferences that affect both the street and transit routing.
+    """
+    accessibility: AccessibilityPreferencesInput
+}
+
+"""
+Stop, station, a group of stop places or multimodal stop place that should be used as
+a location for the search. The trip doesn't have to use the given stop location for a
+transit connection as it's possible to start walking to another stop from the given
+location. If a station or a group of stop places is provided, a stop that makes the most
+sense for the journey is picked as the location within the station or group of stop places.
+"""
+input PlanStopLocationInput {
+    """
+    ID of the stop, station, a group of stop places or multimodal stop place. Format
+    should be `FeedId:StopLocationId`.
+    """
+    stopLocationId: String!
+}
+
+"""
+Mode selections for the plan search.
+"""
+input PlanModesInput {
+    """
+    Should only the direct search without any transit be done.
+    """
+    directOnly: Boolean = false
+
+    """
+    Should only the transit search be done and never suggest itineraries that don't
+    contain any transit legs.
+    """
+    transitOnly: Boolean = false
```

Comment:
We should probably have split the transit search(T) and street search(S) in to queries. 
 - T is time dependent and has a search-window, S is most of the time time-shiftable 
 - The S result is always first
 - The S is excluded when paging
 - The only reason to have it together is to use the S to prune the T - but we could probably find other ways to do that.

Reply:
Do you mean two graphql queries or queries within OTP? Rerunning the street routing is indeed unnecessary when paging. We have thought about some caching solutions for this previously.

Is this comment related to @leonardehrenfried 's question about where the penalty should belong to?